### PR TITLE
Feat: Implemented splash screen

### PR DESCRIPTION
**Pull request by rasvanjaya21**
this PR is under the tagline `simple things are the best most of times` ~ quotes by @lorenzovngl

### CODE REVIEW

**Comment by rasvanjaya21** [comment]
### preview
https://github.com/lorenzovngl/FoodExpirationDates/assets/55085697/d4d37dc8-d459-4635-960f-1b43212f8613

File: app/src/main/res/drawable/ic_splash.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layer-list xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:height="118dp"
+        android:width="118dp"
```
Comment (lorenzovngl) [inline app/src/main/res/drawable/ic_splash.xml:5]: It's better to set it to 100dp x 100dp. I found that on Pixel 3a API 33 the current sizes (118dp) cause a crop.
The guidelines suggests to use a vector icon, but a this time I don't have it.

File: app/src/main/java/com/lorenzovainigli/foodexpirationdates/viewmodel/ExpirationDatesViewModel.kt [hunk 2]
```diff
@@ -17,9 +21,15 @@ class ExpirationDatesViewModel @Inject constructor(
...
 
+    private val _isSplashScreenLoading: MutableState<Boolean> = mutableStateOf(value = true)
+    val isSplashScreenLoading: State<Boolean> = _isSplashScreenLoading
+
     fun getDates(): Flow<List<ExpirationDate>> {
         viewModelScope.launch {
+            _isSplashScreenLoading.value = true
             expirationDates = repository.getAll()
+            delay(2000)
+            _isSplashScreenLoading.value = false
```
Comment (lorenzovngl) [inline app/src/main/java/com/lorenzovainigli/foodexpirationdates/viewmodel/ExpirationDatesViewModel.kt:32]: Good. This codes added an additional delay to the call `repository.getAll()`, so I think it's better to use the delay to execute the call to the database. I propose this code:

```kotlin
val deferred = async {
    expirationDates = repository.getAll()
}
delay(1000)
deferred.await()
```
Here the call to the database is executed asynchronously, and the delay don't wait the db call to terminate. However, if the call to the db is longer that the delay, the `await()` tells the app to don't visualize the list of item unit the db call returned.
Also I think 1 second delay is enough.
Do you agree?

**Review by lorenzovngl — CHANGES_REQUESTED** [review]
That's a good solution! A few refinements and it will be ok.

File: app/src/main/res/drawable/ic_splash.xml (hunk 1 above)
Comment (rasvanjaya21) [reply app/src/main/res/drawable/ic_splash.xml:5]: sorry for late response, 
I got them, I'm forgot to test the responsive for any screen, thanks  for suggestion

File: app/src/main/java/com/lorenzovainigli/foodexpirationdates/viewmodel/ExpirationDatesViewModel.kt (hunk 2 above)
Comment (rasvanjaya21) [reply app/src/main/java/com/lorenzovainigli/foodexpirationdates/viewmodel/ExpirationDatesViewModel.kt]: awesome, I have updated the code as your great advice. strongly agreed. 

tested device for splash screen area (asset not overlapped) :
redmi 9 A12 ✅ | emulator Pixel 5 A11 ✅ | other devices ❎

**Review by lorenzovngl — APPROVED** [review]

**Comment by lorenzovngl** [comment]
Great work! Thank you!

**Pull request merged**